### PR TITLE
OSSM-2153: adds prow jobs for maistra/istio 2.4

### PR DESCRIPTION
**Pull request by bartoszmajsak**
- defines config for maistra-2.4 jobs
- defines maistra-2.4 jobs

### CODE REVIEW

**Comment by bartoszmajsak** [comment]
Awaiting https://github.com/openshift/release/pull/33496

File: ci-operator/config/maistra/api/maistra-api-maistra-2.4.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  commands: |
+    make gen-check \
+    GOBIN="/gobin" \
+    GOCACHE="/tmp/cache" \
+    GOMODCACHE="/tmp/cache" \
+    XDG_CACHE_HOME="/tmp/cache"
+  container:
+    from: src
+zz_generated_metadata:
+  branch: maistra-2.4
```
Comment (yxun) [inline ci-operator/config/maistra/api/maistra-api-maistra-2.4.yaml:34]: @bartoszmajsak This line is referring the source code repo branch.
I am checking and we haven't created branch maistra-2.4 in https://github.com/maistra/api/branches
That's why the CI job failed to pull source code.

File: ci-operator/config/maistra/istio/maistra-istio-maistra-2.4.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  container:
+    from: src
+- as: istio-gencheck-2-4
+  commands: |
+    make gen-check \
+    BUILD_WITH_CONTAINER="0" \
+    GOBIN="/gobin" \
+    GOCACHE="/tmp/cache" \
+    GOMODCACHE="/tmp/cache" \
+    GOFLAGS="-mod=vendor" \
```
Comment (jwendell) [inline ci-operator/config/maistra/istio/maistra-istio-maistra-2.4.yaml:47]: and this

Comment (yxun) [reply ci-operator/config/maistra/istio/maistra-istio-maistra-2.4.yaml:47]: We may not need them when we have   use_build_cache: true in the config yaml.
In the past, I had job failures without those Go env variables before I set the   use_build_cache: true.
We need to test if we still have some failures without Go env variables running in an openshift pod.

**Comment by bartoszmajsak** [comment]
Closing in favor of more granular PRs (such as #33616)

**Pull request closed without merging**